pkg/customevents/events.go

```diff
@@ -0,0 +1,78 @@
+package customevents
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+)
+
+// EventReporter is custom events reporter type which allows user to limit the events
+type EventReporter struct {
+	mux            sync.Mutex
+	recorder       record.EventRecorder
+	reportedEvents map[string]*eventObject
+
+	// report events x times where x is count
+	count int
+
+	// report events after x minutes
+	eventReportAfterMinutes int
+}
+
+// NewEventReporter returns EventReporter object
+func NewEventReporter(recorder record.EventRecorder, count, reportAfter int) *EventReporter {
```

Comment:
If possible think of expanding `count` to `maxCount` or something similar to make it known to the user that it is the max repetition for a particular event.

Reply:
Done